### ISSUE TITLE
Fatal error accessing QuickBooks settings on CiviCRM 4.7

### ISSUE THREAD

**Issue by mlutfy** [agileware/au.com.agileware.civiquickbooks#2]
While testing on CiviCRM 4.7.latest, I received this fatal error when accessing the main QuickBooks settings:

```
Exception: "Cannot determine default entity. CRM_Civiquickbooks_Form_Settings should implement getDefaultEntity()."

#0 /var/aegir/platforms/civicrm-4.7/sites/all/modules/civicrm/CRM/Core/Form.php(1293): CRM_Core_Form->getDefaultEntity()
#1 /var/aegir/platforms/civicrm-4.7/vendor/civicrm/au.com.agileware.civiquickbooks/CRM/Civiquickbooks/Form/Settings.php(34): CRM_Core_Form->addSelect("account_sync_push_contribution_status", "Push Contribution Status")
#2 /var/aegir/platforms/civicrm-4.7/sites/all/modules/civicrm/CRM/Core/Form.php(564): CRM_Civiquickbooks_Form_Settings->buildQuickForm()
#3 /var/aegir/platforms/civicrm-4.7/sites/all/modules/civicrm/CRM/Core/QuickForm/Action/Display.php(92): CRM_Core_Form->buildForm()
#4 /var/aegir/platforms/civicrm-4.7/sites/all/modules/civicrm/packages/HTML/QuickForm/Controller.php(203): CRM_Core_QuickForm_Action_Display->perform(Object(CRM_Civiquickbooks_Form_Settings), "display")
#5 /var/aegir/platforms/civicrm-4.7/sites/all/modules/civicrm/packages/HTML/QuickForm/Page.php(103): HTML_QuickForm_Controller->handle(Object(CRM_Civiquickbooks_Form_Settings), "display")
#6 /var/aegir/platforms/civicrm-4.7/sites/all/modules/civicrm/CRM/Core/Controller.php(351): HTML_QuickForm_Page->handle("display")
#7 /var/aegir/platforms/civicrm-4.7/sites/all/modules/civicrm/CRM/Utils/Wrapper.php(113): CRM_Core_Controller->run()
#8 /var/aegir/platforms/civicrm-4.7/sites/all/modules/civicrm/CRM/Core/Invoke.php(282): CRM_Utils_Wrapper->run("CRM_Civiquickbooks_Form_Settings", "QuickBooks Online Settings", (Array:0))
#9 /var/aegir/platforms/civicrm-4.7/sites/all/modules/civicrm/CRM/Core/Invoke.php(84): CRM_Core_Invoke::runItem((Array:13))
#10 /var/aegir/platforms/civicrm-4.7/sites/all/modules/civicrm/CRM/Core/Invoke.php(52): CRM_Core_Invoke::_invoke((Array:3))
#11 /var/aegir/platforms/civicrm-4.7/sites/all/modules/civicrm/drupal/civicrm.module(448): CRM_Core_Invoke::invoke((Array:3))
#12 /var/aegir/platforms/civicrm-4.7/includes/menu.inc(527): civicrm_invoke("quickbooks", "settings")
#13 /var/aegir/platforms/civicrm-4.7/index.php(21): menu_execute_active_handler()
#14 {main}
```

This setting is from AccountSync, but I could not figure out how `getDefaultEntity()` is supposed to work.

I worked around it with this unclean hack:

```
  function buildQuickForm() {
    [...]

    foreach ($settings as $name => $setting) {
      if (isset($setting['quick_form_type'])) {
        $add = 'add' . $setting['quick_form_type'];
        if ($add == 'addElement') {
          $this->$add($setting['html_type'], $name, $setting['title'], CRM_Utils_Array::value('html_attributes', $setting, array ()));
        }
        // ADDED THE 'elseif' BELOW:
        elseif ($add == 'addSelect' && !empty($setting['pseudoconstant']['optionGroupName'])) {
          $options = CRM_Core_OptionGroup::values($setting['pseudoconstant']['optionGroupName'], FALSE, FALSE, TRUE);
          $this->addElement('select', $name, $setting['title'], $options);
        }
        else {
          $this->$add($name, $setting['title']);
        }

        $description[$name] = $setting['description'];
      }
    }

```

**Comment by jusfreeman** [comment]
Thanks for the bug report @mlutfy - we've got a similar one on our board to fix.

**Comment by agileware-fj** [comment]
@mlutfy This is fixed in the 1.1 release.
